packages/graphql-client/src/graphqlClient.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  SignedTransaction as gqlSignedTransaction,
+  Header,
+  QueryBoxesArgs
+} from "@ergo-graphql/types";
+import {
+  BlockHeader,
+  BoxQuery,
+  BoxWhere,
+  ChainClientBox,
+  HexString,
+  IChainDataClient,
+  NotSupportedError,
+  SignedTransaction
+} from "@fleet-sdk/common";
+import { Client, createClient, fetchExchange, gql } from "@urql/core";
+import { castSignedTxToGql } from "./utils";
+
+export type GraphQLBoxWhere = BoxWhere & {
+  /** Base16-encoded BoxIds */
+  boxIds?: HexString[];
+
+  /** Base16-encoded ErgoTrees or Base58-encoded addresses */
+  contracts?: HexString[];
+};
+
+export type GraphQLBoxQuery = BoxQuery<GraphQLBoxWhere>;
+
+export class GraphqlClient implements IChainDataClient<BoxWhere> {
+  private _client: Client;
+
+  constructor(graphqlUrl: string) {
+    this._client = createClient({
+      url: graphqlUrl,
+      exchanges: [fetchExchange],
+      requestPolicy: "network-only"
+    });
+  }
+  async getUnspentBoxes(q: GraphQLBoxQuery): Promise<ChainClientBox[]> {
+    const query = gql<{ boxes: Omit<ChainClientBox, "confirmed">[] }, QueryBoxesArgs>`
+      query boxes(
+        $boxId: String
+        $boxIds: [String!]
+        $contract: String
+        $contracts: [String!]
+        $template: String
+        $tokenId: String
+        $skip: Int
+        $take: Int
+      ) {
+        boxes(
+          boxId: $boxId
+          boxIds: $boxIds
+          ergoTree: $contract
+          ergoTrees: $contracts
+          ergoTreeTemplateHash: $template
+          tokenId: $tokenId
+          skip: $skip
+          take: $take
+        ) {
+          boxId
+          transactionId
+          index
+          value
+          creationHeight
+          ergoTree
+          assets {
+            tokenId
+            amount
+          }
+          additionalRegisters
+        }
+      }
+    `;
+
+    const response = await this._client.query(query, {
+      boxId: q.where.boxId,
+      boxIds: q.where.boxIds,
+      address: q.where.contract,
+      addresses: q.where.contracts,
+      ergoTreeTemplateHash: q.where.template,
+      tokenId: q.where.tokenId,
+      skip: q.skip,
+      take: q.take,
+      spent: false
+    });
```

Comment:
Let's keep query data concise by avoiding redundant query params like `boxId` and `boxIds`, you can do something like `boxIds: q.where.boxIds ?? q.where.boxId ? [q.where.boxId] : undefined`. Same for `ergoTree`, which is missing here.

Reply:
Done!